lib/src/cli/install/install_command.dart

```diff
@@ -205,36 +194,24 @@ class InstallCommand extends Command<void> {
     await downloadAndExtractBinaries(binaryPath, realmPubspec, archiveName);
 
     print("Realm install command finished.");
+    exit(0);
   }
 
   void validateOptions() {
-    if (options.targetOsType == null && options.packageName == PackageNames.flutter) {
-      abort("Invalid target OS: null.");
-    } else if (options.targetOsType == null && options.packageName == PackageNames.dart) {
-      options.targetOsType = getTargetOS();
-    }
-
-    if (!PackageNames.all.contains(options.packageName)) {
-      abort("Invalid package name: ${options.packageName}");
-    }
-
-    if ((options.targetOsType == TargetOsType.ios || options.targetOsType == TargetOsType.android) && packageName != PackageNames.flutter) {
-      throw Exception("Invalid package name $packageName for target OS ${TargetOsType.values.elementAt(options.targetOsType!.index).name}");
-    }
-  }
-
-  TargetOsType getTargetOS() {
-    if (Platform.isWindows) {
-      return TargetOsType.windows;
-    } else if (Platform.isLinux) {
-      return TargetOsType.linux;
-    } else if (Platform.isMacOS) {
-      return TargetOsType.macos;
+    if (isFlutter) {
+      if (options.targetOsType == null) {
+        abort("Invalid target OS: null.");
+      }
     } else {
-      throw Exception("Unsuported platform ${Platform.operatingSystem}");
+      options.targetOsType ??= getTargetOS();
+      if ((options.targetOsType == TargetOsType.ios || options.targetOsType == TargetOsType.android) && isDart) {
+        throw Exception("Invalid package name $packageName for target OS ${options.targetOsType}");
+      }
     }
   }
```

Comment:
Maybe this could be something like this:
``` dart
void validateOptions() {
    switch (options.flavor) {
      case Flavor.flutter:
        if (options.targetOsType == null) {
          abort("Invalid target OS: null.");
        }
        break;
      case Flavor.dart:
        options.targetOsType ??= getTargetOS();
        if ((options.targetOsType == TargetOsType.ios || options.targetOsType == TargetOsType.android)) {
          throw Exception("Invalid package name $packageName for target OS ${options.targetOsType}");
        }
        break;
    }
  }
  ```